src/CentrifugeRouter.sol

```diff
@@ -45,11 +50,13 @@ contract CentrifugeRouter is Auth, Multicall, ICentrifugeRouter {
     }
 
     // --- Deposit ---
-    function requestDeposit(address vault, uint256 amount) external payable {
-        require(msg.value > 0, "CentrifugeRouter/not-enough-gas-funds");
-        // Due to the multicall nature of this contract, there are issues calling `transfer` multiple times.
-        // Investigate further.
-        payable(gateway).call{value: msg.value}("");
+    function requestDeposit(address vault, uint256 amount, uint256 topUpAmount) external payable {
+        // Assumption: This contract won't hold any ETH
+        require(topUpAmount <= msg.value, "CentrifugeRouter/must-pay-for-tx-costs");
```

Comment:
if we are not using the real estimation from gateway, the user can just send a random topup value.

Reply:
I don't it makes much difference whether they sent  less that it is needed. If it's less than the calculated value then the transaction will fail and they just wasted funds on a failed transaction.

Let me tell you exactly what's happening here. This applies  when there is a multicall transaction.

When the user calls `CentrifugeRouter.multicall([requestDeposit,requestDepost, requestDeposit])`  they send one `msg.value` which should be enough to cover for all three requests. The `topUp` amount is just a fraction of the total amount. You don't want to send all of the `msg.value` to the gateway initially because on the second call the balance of the CentrifugeRouter is `0` even though that `msg.value` is 300 ( 3 x 100 ) for example. It causes issues. 
Let me know if I need to clarify further.

Reply:
ok, cool.
I still think we should work with the exact gas estimation